lib/xenops_server.ml

```diff
@@ -832,14 +832,42 @@ let export_metadata vdi_map vif_map vgpu_pci_map id =
                                         Vm.Indirect { pv_indirect_boot with Vm.devices =
                                                                               List.map (remap_vdi vdi_map) pv_indirect_boot.Vm.devices } } } in
 
+  let diff src dst =
+    let pf = match src.Vgpu.physical_pci_address,
+                   dst.Vgpu.physical_pci_address with
+    | src, dst when src <> dst -> [src, dst]
+    | _ -> [] in
+    let vf = match src.Vgpu.virtual_pci_address,
+                   dst.Vgpu.virtual_pci_address with
+    | Some(src), Some(dst) when src <> dst -> [src, dst]
+    | _ -> [] in
+    List.concat [pf; vf] in
+
+  let pci_map vgpus_src vgpus_dst =
+    List.map2 diff vgpus_src vgpus_dst |> List.concat in
+
+  let remap_pci pci_map pci =
+    let to_string = Pci.string_of_address in
+    match List.assoc_opt pci.Pci.address pci_map with
+    | Some addr ->
+      (* rewrite addr and id (which is constructed from addr *)
+      debug "Remapping PCI %s -> %s"
+        (to_string pci.Pci.address) (to_string addr);
+      let vm = fst pci.Pci.id in
+      let ad = Printf.sprintf "%04x:%02x:%02x.%01x"
```

Comment:
This matches the format in `xapi_xenops.ml`, a bit unfortunate we need to duplicate it here.